### PR TITLE
adr: Adds ADR for project labels

### PR DESCRIPTION
**Pull request by jhmarina**
# Describe your changes

Created ADR explaining the project labels that have been implemented. In the linked issue, there's useful information for reference.

## Issue ticket number and link

https://github.com/aquarist-labs/s3gw/issues/147

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] CHANGELOG.md has been updated should there be relevant changes in this PR.

Signed-off-by: Jesus Herman-Marina <jherman@suse.com>

### CODE REVIEW

File: docs/decisions/0005-project-labels.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
+# Context and Problem Statement
+
+The goal is to put into place standard labels across all SUSE Rancher projects,
+which allows people across multiple spheres of interest a single glance into
+top-level work.
+
+All projects under the SUSE Rancher org are defining standard labels, which will
+allow people across multiple domains of interest a single glance into top-level
+work.
```
Comment (0xavi0) [inline docs/decisions/0005-project-labels.md]: Is this repeating the same concept? 
Also, the first sentence sounds weird to me:
```
which allows people across multiple spheres of interest a single glance into top-level work
```

**Review by m-ildefons — DISMISSED** [review]
LGTM. This is a very sensible change in the face to rancher integration.

File: docs/decisions/0005-project-labels.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+prioritize every ticket since triage is a long process for those actively
+participating in it, but it's important to highlight what needs to sorted out
+urgently.
+
+- priority/0 – Blocker: For bugs, this is an issue that requires immediate
+  attention and is a release blocker.
+- priority/1 – Critical/Urgent: For bugs, this is an issue that breaks existing
+  functionality but doesn't affect the primary use of the app and should be
+  addressed in the next release.
+- priority/2 –
```
Comment (0xavi0) [inline docs/decisions/0005-project-labels.md]: Missed text here?

**Review by giubacc — DISMISSED** [review]
good one; approved with Xavi's suggestions

File: docs/decisions/0005-project-labels.md (hunk 2 above)
Comment (jhmarina) [reply docs/decisions/0005-project-labels.md]: missing indeed, thanks for spotting it!

Comment (jhmarina) [reply docs/decisions/0005-project-labels.md]: fixed @0xavi0

File: docs/decisions/0005-project-labels.md (hunk 1 above)
Comment (jhmarina) [reply docs/decisions/0005-project-labels.md]: Thanks for spotting it! It should be fixed now

**Comment by 0xavi0** [comment]
@jhmarina I still see the same content

**Comment by jhmarina** [comment]
> @jhmarina I still see the same content

I forgot to push the changes 🤦

File: docs/decisions/0005-project-labels.md (hunk 1 above)
Comment (jhmarina) [reply docs/decisions/0005-project-labels.md]: I also reworded the text a bit

File: docs/decisions/0005-project-labels.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+This is the "document kind". Each issue **must** have exactly one `kind/*`
+label. The labels `kind/bug` and `kind/enhancement` are being standardized
+across SUSE Rancher projects.
+
+#### `kind/bug`
+
+This issue describes a defect in the software. Sometimes it requires a judgement
+call, but in general, incomplete functionality, or different behaviour from a
+similar project should be labelled as `kind/enhancement` instead.
```
Comment (jecluis) [inline docs/decisions/0005-project-labels.md:74]: what does it mean to be a different behavior from a similar project?

File: docs/decisions/0005-project-labels.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+#### `kind/enhancement`
+
+This issue requests additional (or a change in) functionality. The request
+typically comes from a user or the PO.
+
+#### `kind/epic`
+
+An epic is an umbrella issue to implement a feature during a single release
```
Comment (jecluis) [inline docs/decisions/0005-project-labels.md]: s/single/multiple/ ?

File: docs/decisions/0005-project-labels.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+This label should only be applied to `kind/bug` issues and signifies that some
+functionality that was working in a previous release is now broken.
+
+### `release-note`
+
+This label is used to mark all issues within a milestone that should be
+explicitly mentioned in the release notes.
+
+### `triage/*`
```
Comment (jecluis) [inline docs/decisions/0005-project-labels.md:131]: shouldn't there be a label representing something that has been triaged, so we don't keep going through issues that have already been triaged?

File: docs/decisions/0005-project-labels.md (hunk 3 above)
Comment (jhmarina) [reply docs/decisions/0005-project-labels.md:74]: "If one expects the s3gw to have the same functionality as a competitor, it's not a bug but rather a feature". This was taken from another Rancher project. But it isn't very clear tbh. Let me rephrase it

File: docs/decisions/0005-project-labels.md (hunk 5 above)
Comment (jhmarina) [reply docs/decisions/0005-project-labels.md:131]: I kind of assumed that once triaged, the issue gets its corresponding labels (ie `kind/*`, `area/*` etc). 

But I added a `triage/ready` label and corresponding text. Better be safe than sorry 😄

File: docs/decisions/0005-project-labels.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+the work.
+
+## Considered Options
+
+Labels can be found here: [https://github.com/aquarist-labs/s3gw/labels][3]
+
+### Issue conventions
+
+All issues should be actionable (within a reasonable timeframe). Usage questions
+and design brainstorming should happen in Github discussions.
```
Comment (votdev) [inline docs/decisions/0005-project-labels.md]:
```suggestion
and design brainstorming should happen in GitHub discussions.
```
Please change that in the whole document.

**Review by votdev — CHANGES_REQUESTED** [review]

File: docs/decisions/0005-project-labels.md (hunk 6 above)
Comment (jhmarina) [reply docs/decisions/0005-project-labels.md]: Updated

File: docs/decisions/0005-project-labels.md (hunk 5 above)
Comment (jhmarina) [reply docs/decisions/0005-project-labels.md:131]: Actually, I had already created a label for a very similar use case, `triage/next-candidate`, which represents issues that have been triaged and could potentially make it to a release. I added it to the ADR instead of `triage/ready`

**Comment by jhmarina** [comment]
@votdev @0xavi0 @jecluis mind having a look again?

File: docs/decisions/0005-project-labels.md [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+
+#### `kind/enhancement`
+
+This issue requests additional (or a change in) functionality. The request
+typically comes from a user or the PO.
+
+#### `kind/epic`
+
+An epic is an umbrella issue to implement a feature that can span over one or
+many release cycle(s). It includes a work breakdown as a task list for
```
Comment (jecluis) [inline docs/decisions/0005-project-labels.md]: s/cycle(s)/cycles/

As is, given the phrase, it makes perfect sense to use 'cycles'.

**Review by jecluis — DISMISSED** [review]
Just a minor nit, but lgtm.

File: docs/decisions/0005-project-labels.md (hunk 7 above)
Comment (jhmarina) [reply docs/decisions/0005-project-labels.md]: fixed!

**Review by jecluis — APPROVED** [review]

**Review by votdev — APPROVED** [review]
